app/models/application_record.rb

```diff
@@ -50,6 +50,10 @@ def add_response_data(prompt, response)
     update(data: data << [prompt, response])
     self
   end
+  
+  def questions_hash
+    data.map{|arr| [arr[3], arr[1]]}.select{|arr| [arr[0], arr[1]].all?}.to_h
```

Comment:
Something that makes it clearer what is in each position of the array. Intermediate variables.